test_direct.py

```
@@ -0,0 +1,354 @@
+    
+    relationships = build_relationships()
+    print(f"\n  Relationships built: {len(relationships)}")
+    
+    # Test traversal
+    def traverse(start_char, min_score, max_depth):
+        """Simulate EntityGraph.traverse_graph()"""
+        relevant = {"character": {start_char}, "group": set()}
+        visited = set()
+        current_frontier = {start_char}
+        
+        for depth in range(max_depth):
```

Ruff (0.15.12)

[warning] Loop control variable `depth` not used within loop body

Rename unused `depth` to `_depth`

(B007)